sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexSquidSensor.java

```diff
@@ -87,91 +102,118 @@ public void describe(SensorDescriptor descriptor) {
-      saveFilesComplexityDistribution(context, inputFile, squidFile);
       saveViolations(context, inputFile, squidFile);
     }
   }
 
-  private static void saveMeasures(SensorContext context, InputFile inputFile, SourceFile squidFile) {
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.NCLOC)
-      .withValue(squidFile.getInt(FlexMetric.LINES_OF_CODE))
-      .save();
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.COMMENT_LINES)
-      .withValue(squidFile.getInt(FlexMetric.COMMENT_LINES))
-      .save();
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.CLASSES)
-      .withValue(squidFile.getInt(FlexMetric.CLASSES))
-      .save();
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.FUNCTIONS)
-      .withValue(squidFile.getInt(FlexMetric.FUNCTIONS))
-      .save();
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.STATEMENTS)
-      .withValue(squidFile.getInt(FlexMetric.STATEMENTS))
-      .save();
-    context.<Integer>newMeasure()
-      .on(inputFile)
-      .forMetric(CoreMetrics.COMPLEXITY)
-      .withValue(squidFile.getInt(FlexMetric.COMPLEXITY))
-      .save();
+  private void saveMeasures(SensorContext context, InputFile inputFile, FlexVisitorContext visitorContext) {
+    FileMetrics metrics = new FileMetrics(visitorContext);
+    saveMeasure(context, inputFile, CoreMetrics.NCLOC, metrics.linesOfCode().size());
+    saveMeasure(context, inputFile, CoreMetrics.COMMENT_LINES, metrics.commentLines().size());
+    saveMeasure(context, inputFile, CoreMetrics.CLASSES, metrics.numberOfClasses());
+    saveMeasure(context, inputFile, CoreMetrics.FUNCTIONS, metrics.numberOfFunctions());
+    saveMeasure(context, inputFile, CoreMetrics.STATEMENTS, metrics.numberOfStatements());
+
+    FileLinesContext fileLinesContext = fileLinesContextFactory.createFor(inputFile);
+    metrics.linesOfCode().forEach(line -> fileLinesContext.setIntValue(CoreMetrics.NCLOC_DATA_KEY, line, 1));
+    metrics.commentLines().forEach(line -> fileLinesContext.setIntValue(CoreMetrics.COMMENT_LINES_DATA_KEY, line, 1));
+    fileLinesContext.save();
+
+    AstNode root = visitorContext.rootTree();
+    int fileComplexity = ComplexityVisitor.complexity(root);
+    saveMeasure(context, inputFile, CoreMetrics.COMPLEXITY, fileComplexity);
+    saveClassComplexity(context, inputFile, root);
+    saveFunctionsComplexityDistribution(context, inputFile, root);
+    saveFilesComplexityDistribution(context, inputFile, fileComplexity);
   }
 
-  private void saveClassComplexity(SensorContext context, InputFile inputFile, SourceFile squidFile) {
-    Collection<SourceCode> classes = scanner.getIndex().search(new QueryByParent(squidFile), new QueryByType(SourceClass.class));
+  private void saveClassComplexity(SensorContext context, InputFile inputFile, AstNode rootNode) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "saveClassComplexity" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-105&issues=AWGd6bADR7xclXvFe_IZ&open=AWGd6bADR7xclXvFe_IZ)

[ISSUE_KEY]: AWGd6bADR7xclXvFe_IZ